new(libscap): gVisor sandboxes trace session set up with runsc

loresuso opened the pull request:
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

> /area driver-kmod

> /area driver-bpf

> /area driver-modern-bpf

> /area libscap-engine-bpf

/area libscap-engine-gvisor

> /area libscap-engine-kmod

> /area libscap-engine-modern-bpf

> /area libscap-engine-nodriver

> /area libscap-engine-noop

> /area libscap-engine-source-plugin

> /area libscap-engine-udig

> /area libscap

> /area libsinsp

> /area tests

> /area proposals

<!--
Please remove the leading whitespace before the `/area <>` you uncommented.
-->

**What this PR does / why we need it**:
This PR introduces the code used to set up new trace sessions for gVisor sandboxes. It does so by leveraging the `runsc` command line tool, which is the standard way to interact with the gVisor runtime. It uses the newly introduced commands for listing sandboxes, and creating and deleting trace sessions. Libs consumers and Falco in particular will have an option to specify the configuration of a trace session. If no file will be specified, we are able to autogenerate it on the fly and unlink it from the filesystem when it will be no more needed. 

**Which issue(s) this PR fixes**:

<!--
Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
If PR is `kind/failing-tests` please post the related issues/tests in a comment and do not use `Fixes`.
-->

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

<!--
If no, you have to do nothing.
If yes, a release note is required:
Delete `NONE` and enter your extended release note in the block below.
Please note, the release note follows the "conventional commit specification" (https://www.conventionalcommits.org/en/v1.0.0/):
For example: `fix: broken link`.
If the PR requires additional action from users switching to the new release, prepend the string "action required:".
For example, `action required: change the API interface of libscap`.
-->

```release-note
NONE
```

Andreagit97 commented:
Thank you @loresuso!
/ok-to-test

poiana commented:
@loresuso: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
build-libs | 7af0b538b77b0d1b0d809f42d7645d85301d0111 | [link](http://prow.falco.org/view/s3/falco-prow-logs/pr-logs/pull/falcosecurity_libs/393/build-libs/1536437812306382848) | true | `/test build-libs`
build-libs-arm | 7af0b538b77b0d1b0d809f42d7645d85301d0111 | [link](http://prow.falco.org/view/s3/falco-prow-logs/pr-logs/pull/falcosecurity_libs/393/build-libs-arm/1536437812360908800) | true | `/test build-libs-arm`

[Full PR test history](http://prow.falco.org/pr-history?org=falcosecurity&repo=libs&pr=393). Please help us cut down on flakes by [linking to](https://git.k8s.io/community/contributors/devel/sig-testing/flaky-tests.md#filing-issues-for-flaky-tests) an [open issue](https://github.com/falcosecurity/libs/issues?q=is:issue+is:open) when you hit one in your PR.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

FedeDP commented on userspace/libscap/engine/gvisor/scap_gvisor.cpp:
```diff
@@ -377,4 +452,161 @@ int32_t engine::next(scap_evt **pevent, uint16_t *pcpuid)
+		return res;
+	}
+
+	int pid = fork();
```
Is this sort of manually re-creating `system()`?
Can't we use `popen` instead?

At the very least, can we use `vfork` here? Fork is super heavy and i do not think that we need to copy everything in the new process just to call exec.

loresuso replied:
Hi Fede! Both `system` and `popen` use a shell to execute the command and I wanted to avoid that. Btw, I agree with you on using `vfork` so I have added it, thank you

LucaGuerra commented on userspace/libscap/engine/gvisor/gvisor.h:
```diff
@@ -102,6 +101,12 @@ class engine {
+    std::vector<std::string> runsc_list();
+    void runsc_trace_create(const std::string &sandbox_id, bool force);
+    void runsc_trace_delete(const std::string &session_name, const std::string &sandbox_id);
+    std::string generate_trace_session_config();
```
I think this function name could be a little better or at least documented. By looking at it it looks like it's generating a configuration string, while it's actually writing a temp file.

LucaGuerra commented on userspace/libscap/engine/gvisor/scap_gvisor.cpp:
```diff
@@ -86,33 +93,53 @@ engine::~engine()
 	unlink(m_socket_path.c_str());
 
-	sock = socket(PF_UNIX, SOCK_SEQPACKET, 0);
+	if(runsc_root_path.empty())
```
I think it'd be great to add a check to verify if the `runsc` binary exists and can be called (we can simply call `runsc --version`) at initialization time. This way we can avoid setting up sockets, threads and so forth if gvisor is not installed.

LucaGuerra commented on userspace/libscap/engine/gvisor/scap_gvisor.cpp:
```diff
@@ -39,6 +43,9 @@ constexpr uint32_t max_ready_sandboxes = 32;
 constexpr size_t initial_event_buffer_size = 32;
 constexpr int listen_backlog_size = 128;
+constexpr size_t max_line_size = 2048;
+// todo(loresuso): change default to k8s path
+const std::string default_runsc_root_path = "/var/run/docker/runtime-runc/moby";
```
We could support a list of default root paths that are queried one after the other unless specified

Andreagit97 commented on userspace/libscap/engine/gvisor/gvisor.h:
```diff
@@ -50,7 +50,6 @@ struct parse_result {
     // pointers to each encoded event within the supplied output buffer
 	std::vector<scap_evt*> scap_events;
 };
-typedef struct parse_result parse_result;
```
Probably we still need this since in `parser.cpp` we use the typedef `parse_result`

LucaGuerra replied:
@loresuso 's point here is that C++ does not need `typedef`s, those are only necessary in C.

Andreagit97 replied:
Oh sorry I didn't know that

Andreagit97 commented on userspace/libscap/engine/gvisor/parsers.cpp:
```diff
@@ -70,7 +70,7 @@ static void fill_context_data(scap_evt *evt, T& gvisor_evt)
 static parse_result parse_container_start(const char *proto, size_t proto_size, scap_sized_buffer scap_buf)
 {
-	struct parse_result ret;
+	parse_result ret;
```
I know this is not strictly necessary but I would love to have this initialized

```suggestion
	parse_result ret = {0};
```

Same for other cases...

Andreagit97 commented on userspace/libscap/engine/gvisor/scap_gvisor.cpp:
```diff
@@ -86,33 +93,53 @@ engine::~engine()
 		return SCAP_FAILURE;
 	}
-
 	unlink(m_socket_path.c_str());
```
Just to remember it:`unlink` syscall doesn't exist in ARM64, maybe we can add a comment here to remember it

FedeDP replied:
Mmmh this is passing build-arm64 ci though.

Btw we **cannot** break arm64 build (that's why we got the CI :D )

Andreagit97 replied:
The `libc` wrapper is there, the problem is that under the hood the `SYS_unlink` syscall doesn't exist in ARM64, so I would avoid calling it, instead, I will call directly  `syscall(SYS_unlinkat, ...); `. This is just my opinion, and must not be done in this PR, but I wrote it here just to remember it. Same for all other syscalls that I have highlighted in this review

Andreagit97 commented on userspace/libscap/engine/gvisor/scap_gvisor.cpp:
```diff
@@ -311,7 +386,7 @@ int32_t engine::process_message_from_fd(int fd)
 int32_t engine::next(scap_evt **pevent, uint16_t *pcpuid)
 {
-	struct epoll_event evts[max_ready_sandboxes];
+	epoll_event evts[max_ready_sandboxes];
```
Do we have a typedef for it somewhere?

loresuso replied:
The struct prefix before using a struct is required only in C, C++ does not have this constraint

FedeDP replied:
Wow didn't know this! 
C++-- .

Andreagit97 commented on userspace/libscap/engine/gvisor/scap_gvisor.cpp:
```diff
@@ -377,4 +452,161 @@ int32_t engine::next(scap_evt **pevent, uint16_t *pcpuid)
+	std::vector<std::string> res;
+	int pipefds[2];
+
+	int ret = pipe(pipefds);
```
same as before, `pipe` is not defined in ARM64

Andreagit97 commented on userspace/libscap/engine/gvisor/scap_gvisor.cpp:
```diff
@@ -377,4 +452,161 @@ int32_t engine::next(scap_evt **pevent, uint16_t *pcpuid)
+	else
+	{
+		::close(pipefds[0]);
+		dup2(pipefds[1], STDOUT_FILENO);
```
`dup2` is not defined in ARM64

Andreagit97 commented on userspace/libscap/engine/gvisor/scap_gvisor.cpp:
```diff
@@ -377,4 +452,161 @@ int32_t engine::next(scap_evt **pevent, uint16_t *pcpuid)
+		int status;
+		
+		::close(pipefds[1]);
+		wait(&status);
```
same as before, `wait` is not defined in ARM64

Andreagit97 commented on userspace/libscap/engine/gvisor/scap_gvisor.cpp:
```diff
@@ -377,4 +452,161 @@ int32_t engine::next(scap_evt **pevent, uint16_t *pcpuid)
+		return res;
+	}
+
+	pid_t pid = vfork();
```
same as before, `vfork` is not defined in ARM64

Andreagit97 commented on userspace/libscap/engine/gvisor/scap_gvisor.cpp:
```diff
@@ -377,4 +452,161 @@ int32_t engine::next(scap_evt **pevent, uint16_t *pcpuid)
+		NULL
+	};
+
+	runsc((char **)argv);
```
Just a stupid question, is there a way to check if the call fails? maybe a return value

Andreagit97 commented on userspace/libscap/engine/gvisor/scap_gvisor.cpp:
```diff
@@ -377,4 +452,161 @@ int32_t engine::next(scap_evt **pevent, uint16_t *pcpuid)
+void engine::runsc_trace_create(const std::string &sandbox_id, bool force)
+{
+	const char *argv[] = {
+		"runsc", 
```
Could these const strings become macros since they are used in different places?

LucaGuerra replied:
Fair point about "runsc" but for now I think it's better to keep it simple. IMO this code is readable as is.

Andreagit97 reviewed (commented):
Just some minor comments, feel free to ignore them if you consider them not necessary. The `arm64` comments are just to remember that we need to change these syscalls in case of support for `arm64`

LucaGuerra commented:
Some thoughts about this:
First of all, I think it's a great PR :rocket: , my comments overall:
- @FedeDP / @Andreagit97 had some concerns about the use of low level calls (`fork`, `unlink`, ...) to manipulate files and run the runsc process. Apparently that's the only way of doing so in vanilla C++. We could either add another dependency to do that (such as [POCO](https://docs.pocoproject.org/current/index.html) ) or leave it like that. I'm good with using low level calls to avoid the burden of adding another dependency.
- We noticed that the interface with `runsc` needs a little tweaking. We need Falco libs to be able to (1) generate a config file/string for the user and (2) load a configuration file supplied by the user. This way it's easier to configure gvisor+falco for new and existing sandboxes. Alas, it might ruin the interface a bit (because the config gen function is not part of the vtable) but we can do what's necessary...

loresuso commented:
In the last commits I have added:
- a way to retrieve errors from runsc, encapsulating the result in a `struct runsc_result`, similarly of what we are doing with the results from the parsers
- a call to `runsc --version` to check if `runsc` tool is correctly installed in the system 
- changed name and behavior of `generate_trace_session_config`, which is now called `print_trace_session_config`. Also added a small comment

I am not 100% convinced about the way we print out the trace session config. Decided for now to hide it behind the vtable, but I am open to suggestions!

Andreagit97 reviewed (dismissed):
/approve

LucaGuerra commented on userspace/libscap/engine/gvisor/scap_gvisor.cpp:
```diff
@@ -86,33 +93,63 @@ engine::~engine()
-	int sock, ret;
-	struct sockaddr_un address;
-	unsigned long old_umask;
+	if(generate_config)
+	{
+		print_trace_session_config();
+		return SCAP_EOF;
+	}
```
I don't think this is the right spot for this call :/
When using Falco, we don't want any initialization to happen before that function is called. Otherwise we would get in output a bunch of log lines and then the json file which is not what we want. I can see two ways out of this:
1. Add it as a separate exported function not part of the engine class (that only exists if gvisor is compiled, of course) which is wired all the way through scap and sinsp
2. Add it straight to sinsp

However, since @Andreagit97 already approved the rest of this PR I would suggest to factor this out of this PR and make another one just for this as the two are independent. WDYT?

poiana commented:
New changes are detected. LGTM label has been removed.

poiana commented:
@loresuso: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
build-libs-minimal | c20c9cc495c9447919a4bd921e964ff86fca2682 | [link](http://prow.falco.org/view/s3/falco-prow-logs/pr-logs/pull/falcosecurity_libs/393/build-libs-minimal/1537410456841359360) | true | `/test build-libs-minimal`

[Full PR test history](http://prow.falco.org/pr-history?org=falcosecurity&repo=libs&pr=393). Please help us cut down on flakes by [linking to](https://git.k8s.io/community/contributors/devel/sig-testing/flaky-tests.md#filing-issues-for-flaky-tests) an [open issue](https://github.com/falcosecurity/libs/issues?q=is:issue+is:open) when you hit one in your PR.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

poiana commented:
@loresuso: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
build-libs-minimal | 6cdc136b2b61da56e8c6f870499fbd42b13e5bb4 | [link](http://prow.falco.org/view/s3/falco-prow-logs/pr-logs/pull/falcosecurity_libs/393/build-libs-minimal/1537442393605804032) | true | `/test build-libs-minimal`

[Full PR test history](http://prow.falco.org/pr-history?org=falcosecurity&repo=libs&pr=393). Please help us cut down on flakes by [linking to](https://git.k8s.io/community/contributors/devel/sig-testing/flaky-tests.md#filing-issues-for-flaky-tests) an [open issue](https://github.com/falcosecurity/libs/issues?q=is:issue+is:open) when you hit one in your PR.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

poiana commented:
@loresuso: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
build-libs | 5c78945f9fe190cdec86985b6f3af9c21d4df590 | [link](http://prow.falco.org/view/s3/falco-prow-logs/pr-logs/pull/falcosecurity_libs/393/build-libs/1537704706191659008) | true | `/test build-libs`

[Full PR test history](http://prow.falco.org/pr-history?org=falcosecurity&repo=libs&pr=393). Please help us cut down on flakes by [linking to](https://git.k8s.io/community/contributors/devel/sig-testing/flaky-tests.md#filing-issues-for-flaky-tests) an [open issue](https://github.com/falcosecurity/libs/issues?q=is:issue+is:open) when you hit one in your PR.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

poiana commented:
@loresuso: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
build-libs | 5c78945f9fe190cdec86985b6f3af9c21d4df590 | [link](http://prow.falco.org/view/s3/falco-prow-logs/pr-logs/pull/falcosecurity_libs/393/build-libs/1537704706191659008) | true | `/test build-libs`
build-libs-arm | 5c78945f9fe190cdec86985b6f3af9c21d4df590 | [link](http://prow.falco.org/view/s3/falco-prow-logs/pr-logs/pull/falcosecurity_libs/393/build-libs-arm/1537704706221019136) | true | `/test build-libs-arm`

[Full PR test history](http://prow.falco.org/pr-history?org=falcosecurity&repo=libs&pr=393). Please help us cut down on flakes by [linking to](https://git.k8s.io/community/contributors/devel/sig-testing/flaky-tests.md#filing-issues-for-flaky-tests) an [open issue](https://github.com/falcosecurity/libs/issues?q=is:issue+is:open) when you hit one in your PR.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

loresuso commented:
/retest build-libs-with-chisels

poiana commented:
@loresuso: The `/retest` command does not accept any targets.
The following commands are available to trigger required jobs:
* `/test build-libs`
* `/test build-libs-arm`
* `/test build-libs-bundled-deps`
* `/test build-libs-minimal`
* `/test build-libs-with-chisels`

Use `/test all` to run all jobs.


<details>

In response to [this](https://github.com/falcosecurity/libs/pull/393#issuecomment-1158627288):

>/retest build-libs-with-chisels


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

loresuso commented:
/test build-libs-with-chisels

loresuso commented:
This PR should be now ready to be merged. I decided that the generation on the fly of the trace session for gVisor will be implemented in the next one. Check it out again and let me know if it is ok, all comments should have been addressed! @LucaGuerra @FedeDP @Andreagit97

FedeDP commented on userspace/libscap/engine/gvisor/CMakeLists.txt:
```diff
@@ -21,6 +23,7 @@ find_package(Threads)
 target_link_libraries(scap_engine_gvisor 
     ${CMAKE_THREAD_LIBS_INIT}
     ${PROTOBUF_LIB}
+    ${JSONCPP_LIB}
```
This feels so weird, i mean compiling the `JSONCPP_LIB_SRC` or linking the `JSONCPP_LIB`.
We should really address it in a subsequent PR :)

loresuso replied:
This is because of the jsoncpp module we currently have! The next PR will address it

FedeDP replied:
Yep i well know it sadly :D

FedeDP commented on userspace/libscap/engine/gvisor/parsers.cpp:
```diff
@@ -70,7 +70,7 @@ static void fill_context_data(scap_evt *evt, T& gvisor_evt)
 static parse_result parse_container_start(const char *proto, size_t proto_size, scap_sized_buffer scap_buf)
 {
-	struct parse_result ret;
+	parse_result ret = {0};
```
I am a fan of this :) 
Thanks!

FedeDP commented on userspace/libscap/engine/gvisor/scap_gvisor.cpp:
```diff
@@ -221,10 +234,67 @@ static void accept_thread(int listenfd, int epollfd)
+
+
+	// Catch all sandboxes that might have been created in the meantime
+	runsc_result new_sandboxes_res = runsc_list();
```
Is this sort of never-ending?
I mean, what if while we run the `new_sandboxes` check, new sandboxes are created?
If we have no way to guarantee that no sandboxes are lost, i think we can skip this check, WDYT?

LucaGuerra replied:
> What if while we run the new_sandboxes check, new sandboxes are created?

In that case they will connect to the socket because it is already been created. The problem is only if new sandboxes are created _before_ the socket is up.

loresuso replied:
Exactly!

FedeDP commented on userspace/libscap/engine/gvisor/scap_gvisor.cpp:
```diff
@@ -239,6 +309,25 @@ int32_t engine::stop_capture()
 	::close(m_epollfd);
 	free_sandbox_buffers();
+
+	runsc_result sandboxes_res = runsc_list();
```
Can't we instead cache the sandboxes list in `start_capture`?
i mean, we could be destroying sessions for unmanaged sandboxes here? Perhaps i miss some key pieces :)

loresuso replied:
I believe caching doesn't work here, other sandboxes may be created after starting the capture and they will all be configured to connect to our socket, so each of them will have a session that must be deleted, if I understand correctly the question

FedeDP replied:
Makes sense; as i told you, i am missing some core key pieces about gvisor inner workings :)

LucaGuerra reviewed (approved):
LGTM!

poiana commented:
@LucaGuerra: changing LGTM is restricted to collaborators

<details>

In response to [this](https://github.com/falcosecurity/libs/pull/393#pullrequestreview-1010236360):

>LGTM!


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

FedeDP commented on userspace/libscap/engine/gvisor/scap_gvisor.cpp:
```diff
@@ -377,4 +466,131 @@ int32_t engine::next(scap_evt **pevent, uint16_t *pcpuid)
+		
+		::close(pipefds[1]);
+		wait(&status);
+		if(status)
```
Shouldn't we be using something like `if (!WIFEXITED(status) || WEXITSTATUS(status) != 0)`.
I mean, we should check:
* if the command was killed by a signal
* if the command exited normally but with a non-zero return status

loresuso replied:
Agreed and added

FedeDP commented on userspace/libscap/scap_open.h:
```diff
@@ -86,6 +86,8 @@ typedef struct scap_open_args
 	bool udig; ///< If true, UDIG will be used for event capture.
 	const char *gvisor_socket; ///< If not null, gvisor will be used for event capture.
+	const char *gvisor_root_path; ///< When using gvisor, the root path used by runsc commands
+	const char *gvisor_trace_session_path; ///< When using gvisor, the trace session config file
```
I think it might be worth to create a struct around these; yet i have no strong opinions on doing that now.

LucaGuerra replied:
Also, we might want to tweak them a little when adding this to Falco. I'm ok with that but I understand your comment.

loresuso replied:
I think we can refactor this when we are sure of all the things that are actually needed. For instance, in the next PR the `gvisor_socket` may be removed from here, because it can be retrieved from the config we will take as input, so my advice is to wait until we have a more precise shape

FedeDP reviewed (commented):
All in all, this LGTM; left some comments.

FedeDP reviewed (approved):
/approve

poiana commented:
LGTM label has been added.  <details>Git tree hash: 7f173d98ffd9e3ddaeef5753f0df7ba000a8cf9e</details>

poiana commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/libs/pull/393#pullrequestreview-1008132576" title="Approved">Andreagit97</a>*, *<a href="https://github.com/falcosecurity/libs/pull/393#pullrequestreview-1010357581" title="Approved">FedeDP</a>*, *<a href="https://github.com/falcosecurity/libs/pull/393#" title="Author self-approved">loresuso</a>*

The full list of commands accepted by this bot can be found [here](https://prow.falco.org/command-help?repo=falcosecurity%2Flibs).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/falcosecurity/libs/blob/master/OWNERS)~~ [Andreagit97,FedeDP]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->

poiana commented:
New changes are detected. LGTM label has been removed.

Andreagit97 commented:
/approve

poiana commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/libs/pull/393#issuecomment-1158806758" title="Approved">Andreagit97</a>*, *<a href="https://github.com/falcosecurity/libs/pull/393#pullrequestreview-1010357581" title="Approved">FedeDP</a>*, *<a href="https://github.com/falcosecurity/libs/pull/393#" title="Author self-approved">loresuso</a>*

The full list of commands accepted by this bot can be found [here](https://prow.falco.org/command-help?repo=falcosecurity%2Flibs).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/falcosecurity/libs/blob/master/OWNERS)~~ [Andreagit97,FedeDP]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->

Andreagit97 reviewed (approved):
/approve

poiana commented:
LGTM label has been added.  <details>Git tree hash: c9e8b6b63fe7cb87eaf64b799bd85f7c51b6c551</details>

poiana commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/libs/pull/393#pullrequestreview-1010410860" title="Approved">Andreagit97</a>*, *<a href="https://github.com/falcosecurity/libs/pull/393#pullrequestreview-1010357581" title="Approved">FedeDP</a>*, *<a href="https://github.com/falcosecurity/libs/pull/393#" title="Author self-approved">loresuso</a>*

The full list of commands accepted by this bot can be found [here](https://prow.falco.org/command-help?repo=falcosecurity%2Flibs).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/falcosecurity/libs/blob/master/OWNERS)~~ [Andreagit97,FedeDP]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->

FedeDP reviewed (approved):
/approve

poiana commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/libs/pull/393#pullrequestreview-1010410860" title="Approved">Andreagit97</a>*, *<a href="https://github.com/falcosecurity/libs/pull/393#pullrequestreview-1010414965" title="Approved">FedeDP</a>*, *<a href="https://github.com/falcosecurity/libs/pull/393#" title="Author self-approved">loresuso</a>*

The full list of commands accepted by this bot can be found [here](https://prow.falco.org/command-help?repo=falcosecurity%2Flibs).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/falcosecurity/libs/blob/master/OWNERS)~~ [Andreagit97,FedeDP]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->
